### PR TITLE
Imroved WordPress page compiler plugin.

### PR DESCRIPTION
**Pull request by felixfontein**
Added support for shortcode plugin data storage.
Adding additional data support to compile_to_string and compile_html.
Added basic gallery shortcode plugin.

(Uses extended metadata provided by getnikola/nikola#1888)

### CODE REVIEW

File: v7/wordpress_compiler/wordpress/plugins/wordpress_shortcode_gallery.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    clazz = re.sub('%[a-fA-F0-9][a-fA-F0-9]', '', clazz)
+    # Limit to A-Z,a-z,0-9,_,-
+    clazz = re.sub('[^A-Za-z0-9_-]', '', clazz)
+    return clazz
+
+
+def sanitize_html_text(text, sanitize_quotes=False):
+    text = text.replace('&', '&amp;')
+    text = text.replace('<', '&lt;')
+    text = text.replace('>', '&gr;')
```
Comment (Kwpolska) [inline v7/wordpress_compiler/wordpress/plugins/wordpress_shortcode_gallery.py]: `&gt;`

Comment (felixfontein) [inline v7/wordpress_compiler/wordpress/plugins/wordpress_shortcode_gallery.py]: Thanks, fixed that.

**Comment by felixfontein** [comment]
Whoops, accidentally pushed to the wrong repo. Gotta rename `origin` to something else :/

**Comment by Kwpolska** [comment]
This is actually okay. You don’t really *need* to send PRs for this plugin, the website is updated automatically at midnight UTC and I just glanced through the code without really looking into it.  As long as it works, just push to `master` and call it a day.

PS. please adjust the [plugin version](https://github.com/getnikola/plugins/blob/master/v7/wordpress_compiler/wordpress_compiler.plugin) with changes.

**Comment by felixfontein** [comment]
Ah, good.
Added another small thing and bumped the version number (d8bdd56267b3ab26b714ca090c1a8d8f5be05984).

**Pull request merged**